apps/04TowersOfHanoi.js

```diff
@@ -19,24 +19,47 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+// Removes the last item of the startStack array,
+// and places it on the end of the endStack array
+function movePiece(startStack, endStack) {
+  var pieceBeingMoved = stacks[startStack].pop();
+  stacks[endStack].push(pieceBeingMoved);
 }
 
-function isLegal() {
-  // Your code here
-
+// Does not allow larger blocks to be placed on top of
+// smaller blocks; Also permits any block to be placed on
+// an empty stack
+function isLegal(startStack, endStack) {
+  var startBlock = stacks[startStack][(stacks[startStack].length - 1)];
+  var endBlock = stacks[endStack][(stacks[endStack].length - 1)];
+
+  if (startBlock < endBlock) {
+    return true;
+  }
+  if (stacks[endStack].length === 0) {
+    return true;
+  }
+  else {
+    return false;
+  }
 }
 
+// Checks if either stack b or stack c has 4 blocks,
+// either of which counts as a win
 function checkForWin() {
-  // Your code here
+  if ((stacks['b'].length === 4) || (stacks['c'].length === 4)) {
+    console.log('You Win!')
+    return true;
+  } else {
+    return false;
+  }
 
 }
 
+// Checks if the move is legal, and, if it is, moves the piece
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
-
+  isLegal(startStack, endStack);
```

Comment:
We aren't actually doing what we said we'd do in our comment! We're ALWAYS moving the block instead of only when the move is legal